dao/src/main/java/se/fortnox/reactivewizard/db/transactions/DaoTransactionsImpl.java

```diff
@@ -68,8 +68,9 @@ public DaoTransactionsImpl() {}
 
     private <T> ConnectionScheduler getConnectionScheduler(Collection<Observable<T>> daoCallsCopy) {
         Observable<T> daoObservable = daoCallsCopy.stream()
+            .filter(obs -> ((DaoObservable<T>)obs).getStatementContextSupplier().get().getConnectionScheduler().hasConnectionProvider())
```

Comment:
If you want to, you could do a map and then filter, avoiding duplicated `.getStatementContextSupplier().get().getConnectionScheduler()`